src/Asv.Drones/Pages/FileBrowser/Tree/BrowserItemViewModel.cs

```diff
@@ -1,112 +1,245 @@
+        get;
+        set => SetField(ref field, value);
     }
 
-    public string EditedName
+    public bool EditMode
     {
-        get => _editedName;
-        set => SetField(ref _editedName, value);
+        get;
+        set => SetField(ref field, value);
     }
 
+    public BindableReactiveProperty<string> EditedName { get; set; }
+
     public string? Crc32Hex
     {
-        get => _crc32Hex;
-        protected set => SetField(ref _crc32Hex, value);
+        get;
+        protected set => SetField(ref field, value);
     }
 
     public Crc32Status Crc32Status
     {
-        get => _crc32Status;
-        set => SetField(ref _crc32Status, value);
-    }
+        get;
+        set => SetField(ref field, value);
+    } = Crc32Status.Default;
 
     public FtpEntryType FtpEntryType
     {
-        get => _ftpEntryType;
-        set => SetField(ref _ftpEntryType, value);
+        get;
+        set => SetField(ref field, value);
+    }
+
+    public ReactiveCommand CommitRename { get; }
+
+    private async ValueTask CommitRenameImpl(CancellationToken ct)
+    {
+        var oldName = Name;
+        var oldPath = Path;
+        var newName = string.IsNullOrWhiteSpace(EditedName.Value)
+            ? BrowserNamingPolicy.BlankName
+            : EditedName.Value;
+        var parentDir = BrowserPathRules.ParentDirOf(oldPath, _separator);
+        var isDirectory = FtpEntryType == FtpEntryType.Directory;
+        var newPath = isDirectory
+            ? BrowserPathRules.CombineDir(parentDir, newName, _separator)
+            : BrowserPathRules.CombineFile(parentDir, newName, _separator);
+
+        EditMode = false;
+        if (
+            string.IsNullOrEmpty(newPath)
+            || string.Equals(newName, oldName, StringComparison.Ordinal)
+        )
+        {
+            EditedName.Value = oldName;
+            return;
+        }
+
+        try
+        {
+            await this.ExecuteCommand(
+                CommitRenameCommand.Id,
+                CommandArg.CreateList(
+                    CommandArg.CreateString(oldPath),
+                    CommandArg.CreateString(newPath)
+                ),
+                ct
+            );
+        }
+        catch (Exception ex)
+        {
+            Logger.LogError(
+                ex,
+                "Failed to rename remote item {oldName} to {newName}",
+                oldName,
+                newName
+            );
+            Name = oldName;
+            EditedName.Value = oldName;
+        }
+    }
+
+    public override IEnumerable<IRoutable> GetRoutableChildren()
+    {
+        return [];
+    }
+
+    public async ValueTask<string> RenameAsync(
+        string oldValue,
+        string newValue,
+        CancellationToken ct
+    )
+    {
+        if (string.IsNullOrWhiteSpace(oldValue))
+        {
+            throw new ArgumentException("Old path is empty", nameof(oldValue));
+        }
+
+        if (string.IsNullOrWhiteSpace(newValue))
```

Comment:
```suggestion
ArgumentException.ThrowIfNullOrEmpty(newValue);
```

И с подобными проверками лучше так делать